walnut/app/schemata.go

```diff
@@ -0,0 +1,245 @@
+                },
+                "monitoring_readout_url": {
+                    "type": "string"
+                }
+            }
+        },
+        "vars": {
+            "type": "object",
+            "title": "User variable definitions",
+            "description": "Variable definitions: vars, override defaults, overridden by user vars."
+        },
+        "roles": {
+            "title": "Roles",
+            "description": "A list of child roles.",
+            "type": "array",
+            "properties": {
+                "name": {
+                    "description": "Parametrized name of an iterator role",
+                    "type": "string"
+                },
+                "for": {
+                    "description": "Amount of Memory",
+                    "type": "object",
+                    "properties": {
+                        "begin": {
+                            "title": "Begin",
+                            "type": "number"
+                        },
+                        "end": {
+                            "title": "End",
+                            "type": "number"
+                        },
+                        "var": {
+                            "title": "variable",
+                            "type": "string",
+                            "enum": [
+                                "it"
+                            ]
+                        }
+                    }
+                },
+                "connect": {
+                    "title": "Connect",
+                    "type": "object",
+                    "properties": {
+                        "name": {
+                            "description": "Name of the inbound channel",
+                            "type": "string"
+                        },
+                        "target": {
+                            "description": "The target entry is a string, with some tree walk functions available for traversing the control tree.",
+                            "type": "string"
+                        },
+                        "type": {
+                            "type": "string",
+                            "enum": [
+                                "pub",
+                                "sub",
+                                "push",
+                                "pull"
+                            ]
+                        },
+                        "sndBufSize": {
+                            "title": "Send Buffer Size",
+                            "type": "number"
+                        },
+                        "rcvBufSize": {
+                            "title": "Receive Buffer Size",
+                            "type": "number"
+                        },
+                        "rateLogging": {
+                            "title": "Rate Logging",
+                            "type": "number"
+                        }
+                    }
+                },
+                "task": {
```

Comment:
If possible, a check should ensure that in a role block either `task` exists, or `roles`, but not both.

`load` is required in `task`.

Reply:
Added in 1cc187a